packages/expo-modules-core/ios/Swift/Logging/PersistentFileLog.swift

```diff
@@ -0,0 +1,156 @@
+ * - Clear the file (remove all entries)
+ *
+ */
+public class PersistentFileLog {
+  private static let EXPO_UPDATES_LOG_QUEUE_LABEL = "dev.expo.updates.logging"
+  private static let serialQueue = DispatchQueue(label: EXPO_UPDATES_LOG_QUEUE_LABEL)
+
+  private let category: String
+  private let filePath: String
+
+  public init(category: String) {
+    self.category = category
+    let fileName = "\(PersistentFileLog.EXPO_UPDATES_LOG_QUEUE_LABEL).\(category).txt"
+    self.filePath = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first?.appendingPathComponent(fileName).path ?? ""
+  }
+  /**
+   Read entries from log file
+   */
+  public func readEntries() -> [String] {
+    if _getFileSize() == 0 {
+      return []
+    }
+    do {
+      return try self._readFileSync()
+    } catch {
+      return []
+    }
+  }
+
+  /**
+   Append entry to the log file
+   Since logging may not require a result handler, the handler parameter is optional
+   If no error handler provided, print error to the console
+   */
+  public func appendEntry(entry: String, _ completionHandler: PersistentFileLogCompletionHandler? = nil) {
+    PersistentFileLog.serialQueue.async {
+      self._ensureFileExists()
+      let size = self._getFileSize()
+      if size == 0 {
+        self._appendTextToFile(text: entry)
+      } else {
+        self._appendTextToFile(text: "\n" + entry)
+      }
+      completionHandler?(nil)
+    }
+  }
+
+  /**
+   Filter existing entries and remove ones where filter(entry) == false
+   */
+  public func filterEntries(filter: @escaping PersistentFileLogFilter, _ completionHandler: @escaping PersistentFileLogCompletionHandler) {
+    PersistentFileLog.serialQueue.async {
+      self._ensureFileExists()
+      do {
+        let contents = try self._readFileSync()
+        let newcontents = contents.filter { entry in filter(entry) }
+        try self._writeFileSync(newcontents)
+        completionHandler(nil)
+      } catch {
+        completionHandler(error)
+      }
+    }
+  }
+
+  /**
+   Clean up (remove) the log file
+   */
+  public func clearEntries(_ completionHandler: @escaping PersistentFileLogCompletionHandler) {
+    PersistentFileLog.serialQueue.async {
+      do {
+        try self._deleteFileSync()
+        completionHandler(nil)
+      } catch {
+        completionHandler(error)
+      }
+    }
+  }
+
+  // MARK: - Private methods
+
+  private func _ensureFileExists() {
+    if !FileManager.default.fileExists(atPath: filePath) {
+      FileManager.default.createFile(atPath: filePath, contents: nil)
+    }
+  }
+
+  private func _getFileSize() -> Int {
+    // Gets the file size, or returns 0 if the file does not exist
+    do {
+      let attrs: [FileAttributeKey: Any?] = try FileManager.default.attributesOfItem(atPath: filePath)
+      return attrs[FileAttributeKey.size] as? Int ?? 0
+    } catch {
+      return 0
+    }
+  }
+
+  private func _appendTextToFile(text: String) {
```

Comment:
```suggestion
  private func appendTextToFile(text: String) {
```

(and all other private functions remove underscore)